File: app/src/main/java/net/imatruck/dsmanager/network/SynologyAPIHelper.java
package net.imatruck.dsmanager.network;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import net.imatruck.dsmanager.R;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by marc on 2017-03-24.
 */

public class SynologyAPIHelper {

    public static SynologyAPI getSynologyApi(Context context) {

        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        String server = prefs.getString(context.getString(R.string.pref_key_server), "");

        if (!server.endsWith("/")) {
            server += "/";
        }

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        httpClient.addInterceptor(interceptor);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(server + "webapi/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        return retrofit.create(SynologyAPI.class);
    }

}

File: app/src/main/java/net/imatruck/dsmanager/network/SynologyAPIHelper.kt
package net.imatruck.dsmanager.network

import android.content.Context
import android.preference.PreferenceManager

import net.imatruck.dsmanager.R

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object SynologyAPIHelper {

    fun getSynologyApi(context: Context): SynologyAPI {

        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        var server: String = prefs.getString(context.getString(R.string.pref_key_server), "")

        if (!server.endsWith('/')) {
            server += "/"
        }

        val interceptor = HttpLoggingInterceptor()
        interceptor.level = HttpLoggingInterceptor.Level.BODY

        val httpClient = OkHttpClient.Builder()
        httpClient.addInterceptor(interceptor)

        val retrofit = Retrofit.Builder()
                .baseUrl(server + "webapi/")
                .addConverterFactory(GsonConverterFactory.create())
                .build()

        return retrofit.create<SynologyAPI>(SynologyAPI::class.java)
    }

}
